analyses/cell-type-ewings/exploratory_analysis/08-merged-celltypes.Rmd

```diff
@@ -0,0 +1,694 @@
+
+## Validation of cell type assignments
+
+In the following section we'll show some plots to help validate that we are content with our cell type assignments. 
+These plots will look specifically at expression of a set of key marker genes that we expect to up in the assigned cell types. 
+These markers are found in `references/combined-markers.tsv` and include a smaller set of markers for each of the cell types we have identified. 
+
+```{r}
+# read in file with validation markers 
+validation_markers_df <- readr::read_tsv(validation_markers_file)
+
+# pull out list of genes 
+genes <- validation_markers_df |>
+  dplyr::pull(ensembl_gene_id)
+
+# get individual gene counts for all marker genes 
+gene_cts <- logcounts(sce[genes, ]) |>
+  as.matrix() |>
+  t() |> 
+  as.data.frame()
+gene_cts$barcodes <- rownames(gene_cts)
+
+# get all unique cell types from the final lumped group 
+celltypes_df <- all_info_df |> 
+  dplyr::select(barcodes, final_lumped)
+
+# create a df that has gene expression column and column indicating whether or not the gene is detected in that cell 
+genes_df <- gene_cts |> 
+  tidyr::pivot_longer(!barcodes, names_to = "ensembl_gene_id", values_to = "gene_exp") |> 
+  dplyr::left_join(celltypes_df, by = c("barcodes")) |> 
+  dplyr::left_join(validation_markers_df, by = c("ensembl_gene_id")) |> 
+  dplyr::rowwise() |> 
+  dplyr::mutate(
+    detected = gene_exp > 0 # column indicating if gene is present or not
+  )
+
+# get total number of cells per final annotation group 
+total_cells_df <- genes_df |> 
+  dplyr::select(barcodes, final_lumped) |> 
+  unique() |> 
+  dplyr::count(final_lumped, name = "total_cells")
+
+# get total number of cells each gene is detected in per group 
+# and mean gene expression per group 
+group_stats_df <- genes_df |> 
+  dplyr::group_by(final_lumped, ensembl_gene_id) |>
+  dplyr::summarize(
+    detected_count = sum(detected),
+    mean_exp = mean(gene_exp)
+  )
+
+gene_summary_df <- genes_df |> 
+  # add total cells
+  dplyr::left_join(total_cells_df, by = c("final_lumped")) |> 
+  # add per gene stats
+  dplyr::left_join(group_stats_df, by = c("ensembl_gene_id", "final_lumped")) |> 
+  dplyr::select(gene_symbol, final_lumped, cell_type, total_cells, detected_count, mean_exp) |> 
+  unique() |>
+  dplyr::rowwise() |> 
+  dplyr::mutate(
+    # get total percent
+    percent_exp = (detected_count/total_cells) * 100,
+    # order genes based on cell type they indicate
+    gene_symbol = factor(gene_symbol, levels = validation_markers_df$gene_symbol),
+    final_lumped = forcats::fct_relevel(final_lumped, cell_type_order)
+    
+  ) 
+```
+
+```{r, fig.height=7}
+# filter out low expressed genes
+dotplot_df <- gene_summary_df |> 
+  dplyr::filter(mean_exp > 0, percent_exp > 10)
+
+
+dotplot <- ggplot(dotplot_df, aes(y = forcats::fct_rev(final_lumped), x = gene_symbol, color = mean_exp, size = percent_exp)) +
+  geom_point() +
+  scale_color_viridis_c(option = "magma") +
```

Comment:
Ok yesterday I learned a really fun trick and probably I'm just eager to apply it, but I wonder if we should cap the scale at 2.5 - 3 to get a better look at the small values - https://github.com/AlexsLemonade/scpca-paper-figures/pull/139#discussion_r1943744044

The upper part of the scale really is only used by EWS-low, but we get the idea that that's high either way.

Reply:
I should also say, part of the reason I want to see those lower values too is because I wonder about the `0` threshold you used above for calling proliferative cells and whether it should be a little more stringent. That cell state isn't a huge concern though since we're confident it's at least tumor!